WalletWasabi.Backend/Filters/ExceptionTranslateFilter.cs

```diff
@@ -19,14 +20,16 @@ public override void OnException(ExceptionContext context)
 			WabiSabiProtocolException e => new JsonResult(new Error(
 				Type: ProtocolConstants.ProtocolViolationType,
 				ErrorCode: e.ErrorCode.ToString(),
-				Description: e.Message))
+				Description: e.Message,
+				Data: JsonConvert.SerializeObject(e.ExceptionData, Formatting.None)))
```

Comment:
> why e.ExceptionData instead of e.Data.

We don't really use the `e.Data`, so it can have unexpected behaviours. It's an `IDictionary` and makes it harder to understand how to use it.

> Why not let asp.net to do it automatically?

Good idea, it works.

Reply:
Much better this way. However, we have to take care that we are using a different serialization here than everywhere else in the code where we use Newtonsoft. But this was as before, this PR just added another property to it. In case of problems, the tests will tell us.

Reply:
Nevermind. Adam bumped into problems. It seems like on the backend side we are using .NET JSON but on the client side, we are using Newtonsoft. Seems like it, but I am not sure @lontivero can you verify that? 

https://github.com/molnard/WalletWasabi/blob/master/WalletWasabi.Backend/Filters/ExceptionTranslateFilter.cs#L19

https://github.com/molnard/WalletWasabi/blob/master/WalletWasabi/Tor/Http/Extensions/HttpResponseMessageExtensions.cs#L68

If it is true we must make this consistent (use newtonsoft both sides) otherwise this is a nest of converter-hell.

Reply:
> It seems like on the backend side we are using .NET JSON but on the client side, we are using Newtonsoft.

No. The serializers are configured in the exact same way in both sides. The problem is that the client cannot guess what type to deserialize it to. Adam has to use the `GenericInterfaceJsonConverter`.